lib/Doctrine/DBAL/Schema/SchemaDiff.php

```diff
@@ -162,7 +163,13 @@ protected function _toSql(AbstractPlatform $platform, $saveMode = false)
         }
 
         foreach ($this->changedTables as $tableDiff) {
-            $sql = array_merge($sql, $platform->getAlterTableSQL($tableDiff));
+            foreach ($platform->getAlterTableSQL($tableDiff) as $alterTableSQL) {
+                if (in_array($alterTableSQL, $sql)) {
```

Comment:
I belive this is unsafe. If generated queries need like drop/recreate ID twice, this will break it.

My question is why do we need this PR at all?

Reply:
It's broken in its current state, not just unsafe. There's no migration that requires the same SQL statement to appear more than once. Generating duplicate statements only generates redundancy, or in this case, a broken migration.